TOC-tidb-cloud.md

```diff
@@ -223,12 +223,12 @@
   - Network Access Control
     - TiDB Serverless
       - [Connect via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections-serverless.md)
-      - [TLS Connections to TiDB Serverless](/tidb-cloud/secure-connections-to-serverless-tier-clusters.md)
+      - [TLS Connections to TiDB Serverless](/tidb-cloud/secure-connections-to-serverless-clusters.md)
```

Comment:
Please also fix the [CI errors](https://github.com/pingcap/docs/actions/runs/5367729363/jobs/9738112741?pr=13969) caused by filename change.